### PR TITLE
fix: markdown parsing error(enter key, code block syntax highlighting)

### PR DESCRIPTION
**Pull request by js87zz**
### Please check if the PR fulfills these requirements
- [x] It's the right issue type on the title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has a description of the breaking change

#### Code block syntax highlighting
* as-is
코드블럭_as_is
* to-be
코드블럭_to_be

#### Enter key with all-selection
* as-is
paragraph_as
* to-be
paragraph_to

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: apps/editor/src/markdown/plugins/syntaxHighlight.ts [hunk 1]
```diff
@@ -46,13 +47,33 @@ export function syntaxHighlight({ schema }: Context) {
   });
 }
 
-function removeMark(nodes: MdNode[], newTr: Transaction) {
+function removeCodeBlockBackground(newTr: Transaction, start: MdPos, end: MdPos, schema: Schema) {
+  const skipLines: number[] = [];
+
+  for (let i = start[0] - 1; i < end[0]; i += 1) {
+    const node = newTr.doc.content.child(i);
+    const { codeStart, codeEnd } = node.attrs;
```
Comment (seonim-ryu) [inline apps/editor/src/markdown/plugins/syntaxHighlight.ts:55]: 아래에 `node` 변수가 사용되는 곳이 없어서 한 줄로 줄여도 될 것 같습니다~ 
```js
const { codeStart, codeEnd } = newTr.doc.content.child(i).attrs;
```

File: apps/editor/src/markdown/mdEditor.ts [hunk 2]
```diff
@@ -105,7 +105,14 @@ export default class MdEditor extends EditorBase {
         keymap({
           'Mod-z': undo(),
           'Shift-Mod-z': redo(),
-          ...baseKeymap
+          ...baseKeymap,
+          Enter: (state, dispatch) => {
+            const emptyNode = createParagraph(this.schema, '');
+            const nodes = [emptyNode, emptyNode];
```
Comment (seonim-ryu) [inline apps/editor/src/markdown/mdEditor.ts:111]: `emptyNode`를 2개 생성해서 넣는 이유는 `\n\n`을 처리하기 위해서인가요~?

**Review by seonim-ryu — COMMENTED** [review]
[11/10] 리뷰 완료합니다! 고생하셨어요~

**Review by lja1018 — APPROVED** [review]
[11/10] 리뷰완료
고생하셨습니다!

File: apps/editor/src/markdown/mdEditor.ts (hunk 2 above)
Comment (js87zz) [reply apps/editor/src/markdown/mdEditor.ts:111]: 빈 노드를 두개 넣어야 기존의 노드를 제거하고 개행이 되는 빈노드를 추가하더라구요. 그래서 두개의 노드를 추가하였습니다~!

**Pull request merged**